lib/components/src/navigation/routed_link.js

```diff
@@ -0,0 +1,44 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+
+const LEFT_BUTTON = 0;
+// Cmd/Ctrl/Shift/Alt + Click should trigger default browser behaviour. Same applies to non-left clicks
+const isPlainLeftClick = e =>
+  e.button === LEFT_BUTTON && !e.altKey && !e.ctrlKey && !e.metaKey && !e.shiftKey;
+
+const wrapOnClick = fn => e => (isPlainLeftClick(e) ? e.preventDefault() || fn(e) : false);
+
+export default class RoutedLink extends React.Component {
+  constructor(props, ...rest) {
+    super(...[props, ...rest]);
+
+    const { onClick } = props;
+    this.onClick = onClick ? wrapOnClick(onClick) : undefined;
+  }
+
+  componentWillUpdate({ onClick }) {
+    this.onClick = wrapOnClick(onClick);
+  }
+
+  render() {
+    const { onClick } = this;
+    const { href, children, ...rest } = this.props;
+    return (
+      <a {...{ href, ...rest, onClick }}>
```

Comment:
I like writing things this way.. in the end JSX properties do not do anything different from this:
transpile into an object that gets fed to the componentFn.

But if people feel strongly I can rewrite to make use of JSX attributes..